{
    "CVE_data_meta": {
        "ASSIGNER": "cve@mitre.org",
        "ID": "CVE-2021-3177",
        "STATE": "PUBLIC"
    },
    "affects": {
        "vendor": {
            "vendor_data": [
                {
                    "product": {
                        "product_data": [
                            {
                                "product_name": "n/a",
                                "version": {
                                    "version_data": [
                                        {
                                            "version_value": "n/a"
                                        }
                                    ]
                                }
                            }
                        ]
                    },
                    "vendor_name": "n/a"
                }
            ]
        }
    },
    "data_format": "MITRE",
    "data_type": "CVE",
    "data_version": "4.0",
    "description": {
        "description_data": [
            {
                "lang": "eng",
                "value": "Python 3.x through 3.9.1 has a buffer overflow in PyCArg_repr in _ctypes/callproc.c, which may lead to remote code execution in certain Python applications that accept floating-point numbers as untrusted input, as demonstrated by a 1e300 argument to c_double.from_param. This occurs because sprintf is used unsafely."
            }
        ]
    },
    "problemtype": {
        "problemtype_data": [
            {
                "description": [
                    {
                        "lang": "eng",
                        "value": "n/a"
                    }
                ]
            }
        ]
    },
    "references": {
        "reference_data": [
            {
                "url": "https://bugs.python.org/issue42938",
                "refsource": "MISC",
                "name": "https://bugs.python.org/issue42938"
            },
            {
                "url": "https://github.com/python/cpython/pull/24239",
                "refsource": "MISC",
                "name": "https://github.com/python/cpython/pull/24239"
            },
            {
                "url": "https://python-security.readthedocs.io/vuln/ctypes-buffer-overflow-pycarg_repr.html",
                "refsource": "MISC",
                "name": "https://python-security.readthedocs.io/vuln/ctypes-buffer-overflow-pycarg_repr.html"
            }
        ]
    }
}